### DISCUSSION TITLE
Explain the consensus mechanism

### DISCUSSION THREAD

**Discussion by jzakotnik** [openkfw/TruBudget#1139] [Ideas]
When introducing trubudget, quite often we get asked how consensus works and we basically say "round robin". I think in the doc we should have a bit of a deep dive, what this means. For example if the partner hosts 5 nodes and the donor 1 (or other way round), can they manipulate data?

**Comment by Stezido** [comment]
Agreed, the only documentation about the consesus mechanism is in the [security FAQ](https://github.com/openkfw/TruBudget/blob/3fb47134be700b3f7a61dc5fcc6d4eecb1b8f631/docs/operation-administration/security.md#why-the-choice-of-round-rip-consensus-for-trubudget)

Multichain has a white-paper including detailed information about specific parts of multichain like the consensus mechanism.
It's a little bit hard to find, even if you downloaded the document. 
The white-paper-document can be accessed and downloaded from [here](https://www.multichain.com/download/MultiChain-White-Paper.pdf)
The section with the consensus information is hidden on page 7 "**Mining in MultiChain**"  

Trubudget should include a detailed documentation about the round-robin consensus in its own docs.

**Comment by jzakotnik** [comment]
Yes, the technical documentation is good, but it's tricky to translate to functional implications, e.g. who could run an attack in the TruBudget setup

**Reply by Stezido** [reply]
What if we provide example scenarios to describe possible functional implications?

**Reply by jzakotnik** [reply]
Actually it would be an interesting setup for a hackathon or similar to try to actually modify the data (given all the open source knowledge but not the other keys)

**Reply by daniel-arnauer** [reply]
relates to #417
